src/components/pages/career-details/career-form/career-form.tsx

```diff
@@ -159,8 +175,8 @@ export const CareerForm: React.FC<CareerFormProps> = (props) => {
       if (key === 'documents') {
         for (let i = 0; i < formValues.documents.length; i++) {
           const keyName = `categorised_documents[${i}][file]`;
-          const fileName = formValues.documents[i].name.split('.')[0];
-          const category = formValues.category_select[fileName];
+          const fileId = createFileId(formValues.documents[i].name);
```

Comment:
we need to access the category select by the correct accessor. We changed the file name to a file id where the category select is build but we forget to change it here. Hence we had some undefined which caused the 500.

Her instead of `a_b` (the new file id way of naming things) we still tried to use `a,b` to access the related category. 
![Screen Shot 2022-02-04 at 15 44 48](https://user-images.githubusercontent.com/1701755/152551764-957b7c72-6076-49b7-82ea-ba37d79e3a67.png)

Of course all of that will fall apart if people upload files that result in conflicting file ids like `a,b` vs `a b` as both will get file ids to `a_b` but to be honest, it's an edge case and it's a mess in there so I think it's good enough right now.